### DISCUSSION TITLE
Platform & Device list matching the executable targets ?

### DISCUSSION THREAD

**Discussion by tdavidcl** [AdaptiveCpp/AdaptiveCpp#1093] [General]
Would it be possible to display only the platforms and devices for which we have compiled.

For example such that 
```c++
const auto &Platforms = sycl::platform::get_platforms() ;    
for (const auto &Platform : Platforms) {
    const auto &Devices = Platform.get_devices() ;
    for (const auto &Device : Devices) {
        ...
    }
}
```
list only devices on which an executable can run?

**Comment by illuhad** [comment]
`device::hipSYCL_has_compiled_kernels()` probably does what you want. Note that this is not bullet-proof. It only takes into account knowledge from the current translation unit. In a production project, where kernel images may come from a variety of places (in an extreme case, think of a dynamically loaded shared library), this may return misleading results. In such a scenario though the notion of "devices on which an executable can run" loses its meaning, as it depends on which part of the application you are referring to.
